tests/vm_startup_test.go

```diff
@@ -0,0 +1,162 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+const timeout = 3 * time.Minute
+const pollingInterval = 5 * time.Second
+const testNamespace = "secondary-test"
+const dnsPort = "31111"
+const dnsIP = "127.0.0.1" // Forwarded to the node port - https://github.com/kubevirt/kubevirtci/pull/867
+const domain = "vm.secondary.io"
+
+var _ = Describe("Virtual Machines Startup", func() {
+
+	BeforeEach(func() {
+		By("Creating test namespace")
+		ns := k8sv1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testNamespace,
+			},
+		}
+		err := getClient().Create(context.Background(), &ns)
+		Expect(err).ToNot(HaveOccurred(), "couldn't create namespace for the test")
+	})
+
+	AfterEach(func() {
+		By("Removing all Virtual Machine Instances from the namespace")
+		vmiList := &v1.VirtualMachineInstanceList{}
+		err := getClient().List(context.Background(), vmiList, &client.ListOptions{Namespace: testNamespace})
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, vmiObject := range vmiList.Items {
+			err = getClient().Delete(context.Background(), &vmiObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Eventually(func() int {
+			vmiList := &v1.VirtualMachineInstanceList{}
+			err := getClient().List(context.Background(), vmiList, &client.ListOptions{Namespace: testNamespace})
+			Expect(err).ToNot(HaveOccurred())
+			return len(vmiList.Items)
+
+		}, timeout, pollingInterval).Should(Equal(0), "failed to remove all vm objects")
+	})
+
+	AfterEach(func() {
+		By("Removing all Network AttachmentDefinitions from the namespace")
+		nadList := &networkv1.NetworkAttachmentDefinitionList{}
+		err := getClient().List(context.Background(), nadList, &client.ListOptions{Namespace: testNamespace})
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, nadObject := range nadList.Items {
+			err = getClient().Delete(context.Background(), &nadObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Eventually(func() int {
+			nadList := &networkv1.NetworkAttachmentDefinitionList{}
+			err := getClient().List(context.Background(), nadList, &client.ListOptions{Namespace: testNamespace})
+			Expect(err).ToNot(HaveOccurred())
+			return len(nadList.Items)
+
+		}, timeout, pollingInterval).Should(Equal(0), "failed to remove all nad objects")
+	})
+
+	AfterEach(func() {
+		By("Removing the test namespace")
+		namespace := &k8sv1.Namespace{}
+		err := getClient().Get(context.Background(), types.NamespacedName{Name: testNamespace}, namespace)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			return getClient().Delete(context.Background(), namespace)
+		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace '%s'", testNamespace))
```

Comment:
Better and nicer please to delete once, and wait that get shows it is gone, as done with the rest
not delete in a loop as done now

Reply:
It isn't correct the Delete Delete .. until gone
it is correct to Delete, Get Get until gone imo